arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -454,14 +477,18 @@ public inline fun <A, B> Iterable<A>.traverse(f: (A) -> Option<B>): Option<List<
 }
 
 @Deprecated(
-  "sequenceOption is being renamed to sequence to simplify the Arrow API",
-  ReplaceWith("sequence()", "arrow.core.sequence")
+  "Sequence is being deprecated in favor of the option DSL.",
+  ReplaceWith("let { l -> result<List<A>> { l.bind() } }", "arrow.core.raise.result")
 )
 public fun <A> Iterable<Option<A>>.sequenceOption(): Option<List<A>> =
-  sequence()
+  let { l -> option { l.bind() } }
 
+@Deprecated(
+  "Sequence is being deprecated in favor of the option DSL.",
```

Comment:
This message may be misleading to users reading it, to me it reads as...
`(The) `Sequence` (datatype) is being deprecated in favor of the option DSL.
It should say, `The function ~sequence~ is being deprecated in favor of the option DSL` or similar wording.